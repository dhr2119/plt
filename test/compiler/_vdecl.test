int main() {
    int d;
    return 0;
}