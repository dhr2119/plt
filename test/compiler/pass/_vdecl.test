int main() {
    int d;
    float f;
    vector float [5] v;
    matrix int [7,6] m;
    bool t;
    String s;
    return 0;
}
