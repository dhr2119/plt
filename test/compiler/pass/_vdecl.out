; ModuleID = 'CMAT'

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt.1 = private unnamed_addr constant [4 x i8] c"%f\0A\00"

declare i32 @printf(i8*, ...)

define i32 @main() {
entry:
  %d = alloca i32
  %f = alloca double
  %v = alloca [5 x double]
  %m = alloca [7 x [6 x i32]]
  %t = alloca i1
  %s = alloca i8*
  ret i32 0
}
